CMakeLists.txt

```diff
@@ -0,0 +1,353 @@
+pkg_check_modules(XML REQUIRED "libxml-2.0")
+pkg_get_variable(PURPLE_PLUGINDIR "purple" "plugindir")
+
+
+#
+# C Pidgin plug-in (with public symbol "purple_init_plugin")
+#   plus an equivalent
+# internal C library for tests to link against
+#
+file(GLOB _LURCH_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/src/lurch*.[ch])
+add_library(lurch_internal STATIC ${_LURCH_SOURCES})
+
+if(WIN32)
+    set(_LURCH_LIBRARY_TYPE MODULE)
+else()
+    set(_LURCH_LIBRARY_TYPE SHARED)
+endif()
+
+# NOTE: On Linux, by default CMake builds binaries with one set of RPATH
+#       and then sets a different value at install time.
+#       For "make install-home" we need a binary that already has the right
+#       RPATH set at build time because there is no install phase where CMake
+#       would change RPATH for us.  If we'd use the install-time RPATH
+#       right away, then the tests wouldn't run, at least not with bundled
+#       axc or libomemo, and at least not the test_lurch_loadable one.
+#       So we separate those two things into two distinct binaries on Linux.
+#       On Windows, RPATH is not a topic and so we don't need to even build
+#       that extra binary on Windows.
+if(NOT WIN32)
+    set(_LURCH_HOME_TARGET lurch_home)
+ENDIF()
+
+foreach(_target lurch ${_LURCH_HOME_TARGET})
+    add_library(${_target} ${_LURCH_LIBRARY_TYPE} ${_LURCH_SOURCES})
+    set_target_properties(${_target} PROPERTIES PREFIX "")  # for lurch.so not liblurch.so
+endforeach()
+
+if(WIN32)
+    if(BUILD_SHARED_LIBS)
+        set(_LURCH_OUTPUT_NAME lurch_fully_shared)
+    else()
+        set(_LURCH_OUTPUT_NAME lurch_semi_shared)
+    endif()
+    set_target_properties(lurch PROPERTIES OUTPUT_NAME ${_LURCH_OUTPUT_NAME})
+else()
+    set(_LURCH_OUTPUT_NAME lurch)
+endif()
+
+if(LURCH_INSTALL)
+    install(TARGETS lurch
+        ARCHIVE DESTINATION ${PURPLE_PLUGINDIR}
+        LIBRARY DESTINATION ${PURPLE_PLUGINDIR}
+        RUNTIME DESTINATION ${PURPLE_PLUGINDIR}
+    )
+endif()
+
+if(WIN32)
+    set(_LURCH_INSTALL_HOME_DEPENDS lurch)
+    set(_LURCH_INSTALL_HOME_OUTPUT_NAME ${_LURCH_OUTPUT_NAME})
+else()
+    set(_LURCH_INSTALL_HOME_DEPENDS ${_LURCH_HOME_TARGET})
+    set(_LURCH_INSTALL_HOME_OUTPUT_NAME ${_LURCH_HOME_TARGET})
+endif()
+
+add_custom_target(install-home
+    COMMAND mkdir -p ~/.purple/plugins
+    COMMAND cp -v
+            ${CMAKE_CURRENT_BINARY_DIR}/${_LURCH_INSTALL_HOME_OUTPUT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}
+            ${_LURCH_AXC_BINARY_GLOB}
+            ${_LURCH_OMEMO_BINARY_GLOB}
+            ~/.purple/plugins/
+    DEPENDS ${_LURCH_INSTALL_HOME_DEPENDS}
+)
+
+# NOTE: This is a 1:1 port of a potentially outdated check
+#       for the name of the Pidgin jabber plug-in acrosses operating systems.
+#       If it turns out outdated, please open bug upstream.  Thank you!
+if(EXISTS /etc/redhat-release)
+    set(_LURCH_OS "Red Hat Linux")
+    set(_LURCH_LIBJABBER xmpp)
+elif(EXISTS /etc/SuSE-release)
```

Comment:
CMake Error at CMakeLists.txt:190 (elif):
  Unknown CMake command "elif".

Reply:
`elseif` should be used here: https://cmake.org/cmake/help/v3.25/command/elseif.html

Reply:
@xvitaly now idea how that slipped through, good catch, fix pushed